src/org/exist/xquery/XQueryContext.java

```diff
@@ -1249,6 +1264,23 @@ public DocumentSet getStaticDocs() {
         return docSupplier.apply(getBroker(), getBroker().getCurrentTransaction(), uri).fold(md -> (Sequence)md, pd -> (Sequence)pd);
     }
 
+    /**
+     * Get's a document from the "Available collections" of the
+     * dynamic context.
+     */
+    public @Nullable Sequence getDynamicallyAvailableCollection(final String uri) throws XPathException {
+        if (dynamicCollections == null) {
+            return null;
```

Comment:
just a q: why didn't you choose for Optional ?

Reply:
I could have equally used `Optional` here, but I felt that in this instance `@Nullable` was a better fit.